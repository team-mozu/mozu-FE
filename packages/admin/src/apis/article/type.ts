export type ArticleAddRequest = {
  title?: string;
  description?: string;
  image?: File | string;
};

export type ArticleDetailResponse = {
  id: number;
  title: string;
  description: string;
  image?: string;
  createDate: string;
  deleteYN: boolean;
};

export type ArticleManagementEditRequest = {
  title?: string;
  description?: string;
  image?: File;
  articleId: number;
};

export type ArticleListResponse = {
  id: number;
  title: string;
  date: string;
};
