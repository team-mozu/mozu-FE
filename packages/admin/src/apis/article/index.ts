import { useMutation, useQueryClient } from "@tanstack/react-query"
import { instance, getCookies } from '@configs/util';
import {ArticleManagementEditRequest} from './type'
import {ArticleAddRequest} from './type'
import {ArticleDetailResponse} from './type'

export const articleManagementAdd = () => {
  const accessToken = getCookies<string>("accessToken");


  return useMutation({
    mutationFn: async (addData: ArticleAddRequest) => {
      const formData = new FormData();
      formData.append("title", addData.title || "");
      formData.append("description", addData.description || "");
      if (addData.image) {
        formData.append("image", addData.image);
      }

      return await instance.post("/article/create", formData, {
        headers: {
          Authorization: `Bearer ${accessToken}`,
          "Content-Type": "multipart/form-data",  
        },
      });
    },
    onSuccess: () => console.log("성공"),
    onError: (error) => console.log("error", error),
  });
};



export const articleManagementDel = () => {
  const accessToken = getCookies<string>("accessToken");

  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: async (articleId: number) => {
      return await instance.delete(`/article/delete/${articleId}`, {
        headers: {
          Authorization: `Bearer ${accessToken}`,
        },
      });
    },
    onSuccess: () => {
      console.log("성공"),
      queryClient.invalidateQueries(["articles"]);
    },
    onError: (error) => console.log("error", error),
  });
};


export const articleManagementDetail =  async(articleId: number):Promise<ArticleDetailResponse> => {
  const accessToken = getCookies<string>("accessToken");
  
    return instance.get(`/article/${articleId}`,{
    headers: {
      Authorization: `Bearer ${accessToken}`,
    },
  }).catch((error) => {
    console.log(error)
  })

};




export const articleManagementEdit = () => {
    const accessToken = getCookies<string>("accessToken");

  
  return useMutation({
    mutationFn: async (data: ArticleManagementEditRequest) => {
      const formData = new FormData();
      formData.append("title", data.title || "");
      formData.append("description", data.description || "");
      if (data.image) {
        formData.append("image", data.image);
      }

      return await instance.post(`/article/update/${data.articleId}`, formData, {
        headers: {
          Authorization: `Bearer ${accessToken}`,
          "Content-Type": "multipart/form-data",  
        },
      });
    },
    onSuccess: () => console.log("성공"),
    onError: (error) => console.log("error", error),
  });
};
