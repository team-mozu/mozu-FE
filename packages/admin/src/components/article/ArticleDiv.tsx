import styled from "@emotion/styled";
import { color, font } from "@mozu/design-token";

interface IArticleType {
  title: string;
  date: string;
  onClick: () => void | boolean;
  selected: boolean;
  articleNumber: number;
}

export const ArticleDiv = ({ title, date, onClick, selected, articleNumber }: IArticleType) => {
  return (
    <ArticleDivContiner
      onClick={onClick}
      selected={selected}>
      <div>
        <ArticleNumber>{articleNumber}</ArticleNumber>
        <ArticleTitle selected={selected}>{title}</ArticleTitle>
      </div>
      <DateDiv selected={selected}>{date}</DateDiv>
    </ArticleDivContiner>
  );
};

const ArticleDivContiner = styled.div<{
  selected: boolean;
}>`
  width: 100%;
  padding: 11px 16px;
  display: flex;
  border-bottom: 1px solid #e4e4e7;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  > div {
    display: flex;
    gap: 20px;
  }
  background-color: ${({ selected }) => (selected ? color.orange[50] : "transparent")};
  ${({ selected }) =>
    !selected &&
    `
  &:hover {
    background-color: ${color.zinc[100]};
  }
  `}
`;

const ArticleTitle = styled.p<{
  selected: boolean;
}>`
  max-width: 340px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font: ${font.b2};
  color: ${({ selected }) => (selected ? color.orange[600] : color.black)};
`;

const DateDiv = styled.p<{
  selected: boolean;
}>`
  font: ${font.l2};
  color: ${({ selected }) => (selected ? color.orange[600] : color.zinc[600])};
`;

const ArticleNumber = styled.p`
  text-align: center;
  font-weight: 400;
  font-size: 14px;
  line-height: 18px;
  color: #52525b;
`;
