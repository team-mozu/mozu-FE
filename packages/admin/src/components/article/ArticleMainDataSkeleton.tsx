import styled from "@emotion/styled";
import { color, font } from "@mozu/design-token";
import { Skeleton } from "../../../../design-token/src/theme/Skeleton";

interface IArticleMainDataType {
  title?: string;
  main?: string;
}

export const ArticleMainDataSkeleton = ({
  title,
  main,
}: IArticleMainDataType) => {
  const lines = main ? main.split("\n") : [];

  return (
    <Container>
      <ArticleImgContainer />
      <ContentWrapper>
        <ArticleTitle>
          <p>{title}</p>
        </ArticleTitle>
        <ArticleMain>
          {lines.map((line) =>
            line.trim() === "" ? <br key={line} /> : <p key={line}>{line}</p>
          )}
        </ArticleMain>
      </ContentWrapper>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  gap: 24px;
  flex-direction: column;
  width: 580px;
`;

const ContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 20px;
`;

const ArticleImgContainer = styled(Skeleton)`
  overflow: hidden;
  width: 100%;
  height: 320px;
  border-radius: 16px;
  background-color: ${color.zinc[100]};
  border: 1px solid ${color.zinc[200]};
`;

const ArticleTitle = styled(Skeleton)`
  color: transparent;
  font: ${font.h4};
  width: 70%;
  white-space: normal;
  word-wrap: break-word;
  word-break: keep-all;
`;

const ArticleMain = styled(Skeleton)`
  p {
    font: ${font.b2};
    color: transparent;
  }
`;
