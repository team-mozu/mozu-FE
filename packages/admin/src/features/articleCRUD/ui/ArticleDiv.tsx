import styled from "@emotion/styled";
import { color, font } from "@mozu/design-token";

interface IArticleType {
  title: string;
  date: string;
  onClick: () => void | boolean;
  selected: boolean;
  articleNumber: number;
}

const formatDate = (dateString: string): string => {
  const date = new Date(dateString);
  return date.toLocaleDateString('ko-KR', {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit'
  }).replace(/\./g, '/').replace(/\s/g, '').slice(0, -1);
};

export const ArticleDiv = ({ title, date, onClick, selected, articleNumber }: IArticleType) => {
  return (
    <ArticleDivContiner
      onClick={onClick}
      selected={selected}>
      <div>
        <ArticleNumber>{articleNumber}</ArticleNumber>
        <ArticleTitle selected={selected}>{title}</ArticleTitle>
      </div>
      <DateDiv selected={selected}>{formatDate(date)}</DateDiv>
    </ArticleDivContiner>
  );
};

const ArticleDivContiner = styled.div<{
  selected: boolean;
}>`
  width: 100%;
  padding: 11px 16px;
  display: flex;
  border-bottom: 1px solid #e4e4e7;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  > div {
    display: flex;
    gap: 20px;
  }
  background-color: ${({ selected }) => (selected ? color.orange[50] : "transparent")};
  ${({ selected }) =>
    !selected &&
    `
  &:hover {
    background-color: ${color.zinc[100]};
  }
  `}
`;

const ArticleTitle = styled.p<{
  selected: boolean;
}>`
  max-width: 350px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font: ${font.b2};
  color: ${({ selected }) => (selected ? color.orange[600] : color.black)};

  /* 반응형 조정 */
  @media (max-width: 1440px) {
    max-width: 240px;
  }

  @media (max-width: 1200px) {
    max-width: 200px;
  }

  @media (max-width: 1024px) {
    max-width: 160px;
  }

  @media (max-width: 900px) {
    max-width: 120px;
  }
`;

const DateDiv = styled.p<{
  selected: boolean;
}>`
  font: ${font.l2};
  color: ${({ selected }) => (selected ? color.orange[600] : color.zinc[600])};
`;

const ArticleNumber = styled.p`
  text-align: center;
  font-weight: 400;
  font-size: 14px;
  line-height: 18px;
  color: #52525b;
`;
