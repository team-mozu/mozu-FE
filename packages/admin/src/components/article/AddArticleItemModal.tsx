import { useGetArticleList } from '@/apis';
import styled from '@emotion/styled';
import { color, font } from '@mozu/design-token';
import { Button, SearchInput } from '@mozu/ui';
import { useState, useEffect } from 'react';
import { ArticleItem } from './ArticleItem';

// 서버 응답 인터페이스 정의
interface ArticleItem {
  id: number;
  title: string;
  date: string;
}

interface IArticleModalType {
  close: () => void;
  onArticlesSelected: (articles: any[]) => void;
  selectedDegree: number;
  existingArticles?: { id: number }[];
}

export const AddArticleItemModal = ({
  close,
  onArticlesSelected,
  selectedDegree,
  existingArticles = [],
}: IArticleModalType) => {
  // API에서 기사 리스트 가져오기
  const { data: articleData } = useGetArticleList();
  const [searchText, setSearchText] = useState('');

  const articles = articleData?.article || [];
  const existingArticleIds = existingArticles.map((article) => article.id);

  // 이미 선택된 아이템 필터링 및 검색어 적용
  const filteredArticles = articles.filter(
    (article) =>
      !existingArticleIds.includes(article.id) &&
      (searchText === '' ||
        article.title.toLowerCase().includes(searchText.toLowerCase()) ||
        String(article.id).includes(searchText)),
  );

  const [checkedArticles, setCheckedArticles] = useState<boolean[]>(
    Array(filteredArticles.length).fill(false),
  );

  const [isHeadCheck, setIsHeadCheck] = useState<boolean>(false);

  // 필터링된 아이템이 변경될 때마다 체크 상태 초기화
  useEffect(() => {
    setCheckedArticles(Array(filteredArticles.length).fill(false));
    setIsHeadCheck(false);
  }, [filteredArticles.length, searchText]);

  // 개별 아이템 체크 토글
  const checkClick = (index: number) => {
    setCheckedArticles((prev) => {
      const updateCheckItems = [...prev];
      updateCheckItems[index] = !updateCheckItems[index];

      // 헤더 체크박스 상태 업데이트
      const allChecked = updateCheckItems.every((item) => item);
      setIsHeadCheck(allChecked);

      return updateCheckItems;
    });
  };

  // 선택 완료 후 제출
  const handleSubmit = () => {
    const selectedArticles = filteredArticles
      .filter((_, index) => checkedArticles[index])
      .map((article) => ({
        id: article.id,
        title: article.title,
      }));

    console.log('Selected articles in modal:', selectedArticles);
    onArticlesSelected(selectedArticles);
  };

  // 전체 선택 토글
  const headClick = () => {
    const newState = !isHeadCheck;
    setIsHeadCheck(newState);
    setCheckedArticles(Array(filteredArticles.length).fill(newState));
  };

  // 검색어 변경 처리
  const handleSearchChange = (value: string) => {
    setSearchText(value);
  };

  // 선택된 아이템이 있는지 확인
  const hasSelectedItems = checkedArticles.some((item) => item);

  return (
    <ModalBackground>
      <ArticleContainer>
        <SearchContainer>
          <Title isHeader>기사 추가</Title>
          <SearchInput
            inputText="기사 검색.."
            value={searchText}
            onChange={handleSearchChange}
          />
        </SearchContainer>
        <TableContainer>
          <ArticleItem
            isHeader={true}
            title1="기사 제목"
            title2="등록일자"
            id="title"
            checked={isHeadCheck}
            onChange={headClick}
          />
          <ItemContents>
            {filteredArticles.length > 0 ? (
              filteredArticles.map((article, index) => (
                <ArticleItem
                  title1={article.title}
                  title2={article.date}
                  onChange={() => checkClick(index)}
                  checked={checkedArticles[index]}
                  id={String(article.id)}
                  key={article.id}
                />
              ))
            ) : (
              <EmptyState>
                {searchText
                  ? '검색 결과가 없습니다.'
                  : '추가 가능한 기사가 없습니다.'}
              </EmptyState>
            )}
          </ItemContents>
        </TableContainer>
        <FooterContainer>
          <BtnContainer>
            <Button
              backgroundColor={color.zinc[50]}
              borderColor={color.zinc[200]}
              color={color.zinc[800]}
              onClick={close}
            >
              취소
            </Button>
            <Button
              backgroundColor={color.orange[500]}
              borderColor={color.orange[500]}
              color={color.white}
              onClick={handleSubmit}
              disabled={!hasSelectedItems}
            >
              선택 기사 추가
            </Button>
          </BtnContainer>
        </FooterContainer>
      </ArticleContainer>
    </ModalBackground>
  );
};

// Styled components
const ModalBackground = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.08);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
`;

const ArticleContainer = styled.div`
  width: 800px;
  height: 640px;
  border-radius: 16px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: ${color.white};
  padding-top: 12px;
`;

const SearchContainer = styled.header`
  display: flex;
  flex-direction: column;
  gap: 12px;
  align-items: start;
  width: 776px;
  margin-bottom: 12px;
`;

const TableContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  border-bottom: 1px solid ${color.zinc[200]};
`;

const FooterContainer = styled.footer`
  width: 100%;
  height: 64px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding-right: 12px;
`;

const BtnContainer = styled.div`
  display: flex;
  gap: 10px;
`;

const ItemContents = styled.div`
  overflow: auto;
  height: 432px;
`;

const EmptyState = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 200px;
  font: ${font.b2};
  color: ${color.zinc[500]};
`;

const Title = styled.div<{ isHeader: boolean }>`
  font: ${font.b1};
  color: ${color.black};
  margin-left: 4px;
`;
