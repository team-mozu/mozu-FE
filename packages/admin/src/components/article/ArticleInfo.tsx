import styled from "@emotion/styled";
import { color, font } from "@mozu/design-token";
import { Button } from "@mozu/ui";
import { useRef, useState } from "react";

export const ArticleInfo = () => {
  const [datas, setDatas] =
    useState<
      {
        isClicked: boolean;
        articleContent: [
          {
            title: string;
          },
        ];
      }[]
    >([]);
  const [isClose, setIsClose] = useState<boolean>(false);

  const outSideRef = useRef<HTMLDivElement>(null);
  const outSideClick = (e: React.MouseEvent<HTMLDivElement>) => {
    if (e.target === e.currentTarget) {
      setIsClose(true);
    }
  };

  const cancelClick = () => {
    setIsClose(true);
  };

  const barClick = (index: number) => {
    setDatas(prev =>
      prev.map((data, idx) =>
        idx === index
          ? {
            ...data,
            isClicked: true,
          }
          : {
            ...data,
            isClicked: false,
          },
      ),
    );
  };

  return (
    !isClose && (
      <BackgroundContainer
        onClick={outSideClick}
        ref={outSideRef}>
        <ModalContainer>
          <ContentContainer>
            <TitleContainer>
              <Title>기사 정보</Title>
            </TitleContainer>
            <BarContainer>
              {datas.map((data, index) => (
                <BarContent
                  key={data.articleContent[0].title}
                  isClicked={data.isClicked}
                  onClick={() => barClick(index)}>
                  {index + 1}차
                </BarContent>
              ))}
            </BarContainer>
            <ArticleContainer>
              {datas
                .filter(data => data.isClicked)
                .map((data) =>
                  data.articleContent.map((content) => (
                    <ArticleContent key={content.title}>{content.title}</ArticleContent>
                  )),
                )}
            </ArticleContainer>
            <FooterContainer>
              <Button
                backgroundColor={color.zinc[50]}
                borderColor={color.zinc[200]}
                color={color.zinc[800]}
                onClick={cancelClick}>
                닫기
              </Button>
            </FooterContainer>
          </ContentContainer>
        </ModalContainer>
      </BackgroundContainer>
    )
  );
};

const FooterContainer = styled.div`
  width: 100%;
  border-top: 1px solid ${color.zinc[200]};
  display: flex;
  justify-content: end;
  padding: 12px 12px 12px 0;
`;

const TitleContainer = styled.div`
  padding: 24px 0 12px 24px;
  width: 100%;
`;
const ContentContainer = styled.div`
  width: 100%;
`;

const Title = styled.div`
  font: ${font.t3};
  color: ${color.black};
`;

const ModalContainer = styled.div`
  width: 800px;
  height: 360px;
  border-radius: 16px;
  background-color: ${color.white};
  display: flex;
  justify-content: center;
  align-items: center;
`;

const BackgroundContainer = styled.div`
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.08);
  position: fixed;
  top: 0;
  z-index: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const BarContent = styled.button<{
  isClicked: boolean;
}>`
  width: 80px;
  height: 40px;
  background-color: transparent;
  color: ${({ isClicked }) => (isClicked ? color.orange[500] : color.zinc[700])};
  border-bottom: 1px solid
    ${({ isClicked }) => (isClicked ? color.orange[500] : "none")};
  outline: none;
  display: flex;
  justify-content: center;
  align-items: center;
  font: ${font.t3};
  transition: border-bottom 0.5s ease;
`;

const BarContainer = styled.div`
  width: 100%;
  display: flex;
  border-bottom: 1px solid ${color.zinc[200]};
`;

const ArticleContainer = styled.div`
  width: 100%;
  height: 200px;
  overflow: scroll;
  display: flex;
  flex-direction: column;
  gap: 24px;
  padding: 24px;
`;

const ArticleContent = styled.div`
  font: ${font.b2};
  color: ${color.black};
`;
