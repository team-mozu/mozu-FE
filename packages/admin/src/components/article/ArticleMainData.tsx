import styled from "@emotion/styled";
import { color, font } from "@mozu/design-token";
import { NoNewsImg } from "@mozu/ui";

interface IArticleMainDataType {
  img?: string | null | undefined;
  title?: string;
  main?: string;
}

export const ArticleMainData = ({ img, title, main }: IArticleMainDataType) => {
  const lines = main ? main.split("\n") : [];

  return (
    <Container>
      <ArticleImgContainer>
        {img ? (
          <ArticleImg
            src={img}
            alt="기사 이미지"
          />
        ) : (
          <NoNewsImg />
        )}
      </ArticleImgContainer>
      <ContentWrapper>
        <ArticleTitle>
          <p>{title}</p>
        </ArticleTitle>
        <ArticleMain>
          {lines.map((line) => (line.trim() === "" ? <br key={line} /> : <p key={line}>{line}</p>))}
        </ArticleMain>
      </ContentWrapper>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  gap: 24px;
  flex-direction: column;
  width: 580px;
`;

const ContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 20px;
`;

const ArticleImg = styled.img`
  width: 100%;
  background-color: transparent;
`;

const ArticleImgContainer = styled.div`
  overflow: hidden;
  width: 100%;
  height: 320px;
  border-radius: 16px;
  background-color: ${color.zinc[100]};
  border: 1px solid ${color.zinc[200]};
`;

const ArticleTitle = styled.div`
  color: ${color.black};
  font: ${font.h4};
  width: 70%;
  white-space: normal;
  word-wrap: break-word;
  word-break: keep-all;
`;

const ArticleMain = styled.div`
  p {
    font: ${font.b2};
    color: ${color.zinc[800]};
  }
`;
