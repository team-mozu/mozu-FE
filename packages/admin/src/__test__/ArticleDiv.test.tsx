import { color } from "@mozu/design-token";
import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { ArticleDiv } from '../components/article/ArticleDiv';

describe("ArticleDiv", () => {
  const defaultProps = {
    title: "Test Article",
    date: "2025-08-08",
    onClick: vi.fn(),
    selected: false,
    articleNumber: 1,
  };

  it("renders correctly with all content", () => {
    render(<ArticleDiv {...defaultProps} />);
    expect(screen.getByText("Test Article")).toBeInTheDocument();
    expect(screen.getByText("2025-08-08")).toBeInTheDocument();
    expect(screen.getByText("1")).toBeInTheDocument();
  });

  it("calls onClick when clicked", () => {
    const mockOnClick = vi.fn();
    render(
      <ArticleDiv
        {...defaultProps}
        onClick={mockOnClick}
      />,
    );
    fireEvent.click(screen.getByRole("button"));
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it("applies selected styles when selected", () => {
    render(
      <ArticleDiv
        {...defaultProps}
        selected={true}
      />,
    );
    const container = screen.getByRole("button");
    expect(container).toHaveStyle(`background-color: ${color.orange[50]}`);
  });

  it("matches snapshot", () => {
    const { container } = render(<ArticleDiv {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot when selected", () => {
    const { container } = render(
      <ArticleDiv
        {...defaultProps}
        selected={true}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
