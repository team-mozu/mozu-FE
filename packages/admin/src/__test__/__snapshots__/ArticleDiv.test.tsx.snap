// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArticleDiv > matches snapshot 1`] = `
<div
  class="css-13lfn1v"
  role="button"
  tabindex="0"
>
  <div>
    <p
      class="css-1509n5v"
    >
      1
    </p>
    <p
      class="css-1gn182m"
    >
      Test Article
    </p>
  </div>
  <p
    class="css-xa9rnn"
  >
    2025-08-08
  </p>
</div>
`;

exports[`ArticleDiv > matches snapshot when selected 1`] = `
<div
  class="css-1itllec"
  role="button"
  tabindex="0"
>
  <div>
    <p
      class="css-1509n5v"
    >
      1
    </p>
    <p
      class="css-bmk7cl"
    >
      Test Article
    </p>
  </div>
  <p
    class="css-1h9e9lw"
  >
    2025-08-08
  </p>
</div>
`;
